### PR TITLE
feat: overhaul attacks [wip]

### LINKED ISSUE #214: [Bug] Missing damage types of weapons
### Describe the bug

All weapons are missing their relative damage types

### NodeJS Version

14.17.0

### npm Version

6.14.13

### warframe-items Version

1.1248.9

### Access Method

github download

### What happened?

All of the weapons for some reason doesn't have damage types anymore. I have no ideia in which version this did occur, but both api and repository are missing it.

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? (provide a description or issue closes statement)
I overhauled the attack structure
closes #214 

---

### Reproduction steps
Lots of potential attacks were missing from wikia data

---

### Considerations
- Does this contain a new dependency? **[Yes]**
- Does this introduce opinionated data formatting or manual data entry? **[Yes]**, opinionated data formatting
- Does this pr include the updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter through `npm run lint`? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Enhancement]**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -564,4 +594,18 @@ declare module 'warframe-items' {
...
         '---'
+        
+    type Element = 
+        'Impact' |
+        'Puncture' |
+        'Slash' |
+        'Toxin' |
+        'Electricity' |
+        'Blast' |
+        'Radiation' |
```
Comment (AyAyEm) [inline index.d.ts:605]: What do you think of already adding Gas so we don't have to update the types when a weapon of that type does launch?

Comment (AyAyEm) [reply index.d.ts:605]: Also we can use the DamageTypes interface and just use Element as an alias:
`type Element = Capitalize<keyof DamageTypes>;`

File: index.d.ts [hunk 2]
```diff
@@ -209,6 +209,7 @@ declare module 'warframe-items' {
         heavySlamRadialDamage?: number;
         heavySlamRadius?: number;
         windUp?: number;
+        attacks: (Attack|SlamAttack)[];
```
Comment (AyAyEm) [inline index.d.ts:212]: The majority of items components doesn't have attacks.

Suggestion:
`attacks?: (Attack|SlamAttack)[];`

File: index.d.ts [hunk 3]
```diff
@@ -231,6 +232,35 @@ declare module 'warframe-items' {
...
+        status_chance?: number;
+        charge_time?: number;
+        shot_type?: string;
+        shot_speed?: number;
+        flight?: number | string;
+        falloff?: Falloff;
+        damage: DamageTypes;
+    }
+    
+    interface SlamAttack {
```
Comment (AyAyEm) [inline index.d.ts:253]: Is there a real need for SlamAttack to be seperated from Attack? As there's other attack types which does not have a separated interface for it.

**Review by AyAyEm — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 3 above)
Comment (TobiTenno) [reply index.d.ts:257]: The others don't have radically different formats for the source data

File: index.d.ts (hunk 2 above)
Comment (TobiTenno) [reply index.d.ts:212]: It'll still be an empty array, which would always exist

File: index.d.ts (hunk 1 above)
Comment (TobiTenno) [reply index.d.ts:609]: Feel free to use the code suggestions so I can merge them from here.

Comment (TobiTenno) [reply index.d.ts:609]: Like I've mentioned, I don't use the types, so this is entirely up for your convenience

File: index.d.ts (hunk 3 above)
Comment (TobiTenno) [reply index.d.ts:257]: I would honestly do a melee one, but I don't know how it works for doing things that extend others.

File: index.d.ts (hunk 2 above)
Comment (AyAyEm) [reply index.d.ts:212]: If so then there's a fix to be made because not every component have an empty attacks array.

Comment (TobiTenno) [reply index.d.ts:212]: yeah, i ended up changing it, but yeah, I see your point now, since this is on the Item interface, not a Weapon interface

**Review by SlayerOrnstein — APPROVED** [review]
LGTM

File: config/itemTypes.json [hunk 4]
```diff
@@ -304,4 +304,40 @@
...
+  "name": "Pets"
+}, {
+  "id": "LichToken",
+  "name": "Resource"
+}, {
+  "id": "Items/Deimos/Seriglass",
+  "name": "Resource"
+}, {
+  "id": "ZanukaPets/ZanukaPetParts",
+  "name": "PEt Resource"
```
Comment (Everspace) [inline config/itemTypes.json:342]: Typo?

Comment (TobiTenno) [reply config/itemTypes.json:342]: yep, i'll fix that now

```suggestion
  "name": "Pet Resource"
```

**Pull request merged**